koap/src/commonMain/kotlin/Message.kt

```diff
@@ -291,17 +293,21 @@ sealed class Message {
              * - `0` (register) adds the entry to the list, if not present;
              * - `1` (deregister) removes the entry from the list, if present.
              */
-            enum class Registration { Register, Deregister }
```

Comment:
I'm a little confused by this. I thought `enum`s from Kotlin would work fine in JS?
The `Enum` class itself is marked as supporting JS platform.

https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-enum/

Reply:
This is the error:
> Message.kt: (296, 24): Declaration of such kind (enum class) cant be exported to JS

¯\\_(ツ)_/¯ 

I think it's objecting due to it being nested. I took the minimal route of declaring it as a `sealed class` but an alternative would probably be to move all enums to the root level. Again, probably best left to a separate PR to do that kind of refactoring.

Reply:
Oh, looks like `enum`s just can't be exported _yet_. So I don't think it's a nested issue.

https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.js/-js-export/

---

`sealed class` is a good alternative. 👍 
My original hesitation was I was hoping we could get the JS target going w/o breaking API changes; looks like this will just have to be a major version bump (which is fine).